### PR TITLE
Simple Unit Test & Terminal Color Modules

### PR DESCRIPTION
**Pull request by chrisnevers**
While playing around I found myself writing these modules/functions and figured they could be generally handy? It doesn't seem like Amulet has a package manager yet, with that said I understand if these don't belong in the standard library.

### Colors
* Composable functions for changing the color of terminal text.

### Test
* Some methods for testing functions
```OCaml
run_test   : string * (unit -> bool) -> unit
run_tests  : list (string * (unit -> bool)) -> unit
expect     : eq 'a * show 'a => string * (unit -> 'a) * 'a -> unit
```

#### Example Program
```OCaml
let test_add_1 () = add (Some 1) (Some 4) == 5
let test_add_2 () = add None None == 0

Test.run_tests [
  ("If both are present, add", test_add_1),
  ("If both absent, return 0", test_add_2)
]
Test.run_test ("Quick math", fun () -> 2 + 2 == 4)
Test.expect ("Get 3", fun () -> 5, 3)

```

#### Example Output
Screen Shot 2020-01-31 at 8 05 00 PM

### CODE REVIEW

File: lib/amulet/colors.ml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+open import "./base.ml"
+
+module Colors = struct
```
[the remark this answers is not in the document]
Comment (SquidDev) [reply lib/amulet/colors.ml]: Now that I think about it, might also be worth moving to `lib/io/ansi.ml` instead? I'll wait to see what @plt-abigail thinks.

File: lib/amulet/list.ml [hunk 2]
```diff
@@ -262,3 +262,22 @@ let scanl func q ls =
...
+  | Nil -> []
+  | Cons (None, tl) -> compact tl
+  | Cons (Some a, tl) -> a :: compact tl
+
+(**
+ * length returns the number of elements in the list
+ *
+ * Runtime: O(n)
+ *)
+let rec length = function
```
[the remark this answers is not in the document]
Comment (chrisnevers) [reply lib/amulet/list.ml]: Thanks, it looks like `length` is already implemented in foldable 👍🏼

File: lib/amulet/list.ml [hunk 3]
```diff
@@ -262,3 +262,22 @@ let scanl func q ls =
         go (fun xs -> k (z :: xs)) (func z x) xs
   go (fun x -> x) q ls
 
+(**
+ * compact takes a list of optionals and returns the present elements
+ *
+ * Runtime: O(n)
+ *)
+let rec compact = function
```
[the remark this answers is not in the document]
Comment (chrisnevers) [reply lib/amulet/list.ml]: Done!

**Comment by plt-abigail** [comment]
Please use currying instead of tuples to define functions of multiple arguments.

**Comment by SquidDev** [comment]
I've cherry-picked the filter_map implementation in the above commit. Having had a chat with @plt-abigail, we decided we're not entirely sure the ANSI support. Sorry!

**Pull request closed without merging**